src/app/components/AnalyticsConsent/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+import Snackbar from '@mui/material/Snackbar'
+import Typography from '@mui/material/Typography'
+import { styled } from '@mui/material/styles'
+import Button from '@mui/material/Button'
+import Link from '@mui/material/Link'
+import { Trans, useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import CardActions from '@mui/material/CardActions'
+import { useScreenSize } from 'app/hooks/useScreensize'
+import * as matomo from './initializeMatomo'
+import { legalDocuments } from '../../utils/externalLinks'
+
+matomo.loadMatomo()
+
+const AcceptCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+const DeclineCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+export const AnalyticsConsent = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useScreenSize()
+  const [hasAccepted, setHasAccepted] = useState<matomo.HasAccepted>(matomo.hasAccepted())
+
+  useEffect(() => {
+    window._paq.push([
+      function () {
+        // Matomo loaded
+        setHasAccepted(matomo.hasAccepted())
+        // TODO: bug: still sometimes gets stuck with 'error_matomo_not_loaded' ???
+        console.log('loaded', matomo.hasAccepted())
```

Comment:
In order to be sure that matomo has loaded, onload callback should be added to the script. Since we are dynamically loading it.

One thing that could be helpful is to use useEffectLayout instead of useEffect.

Reply:
Instead of onload I'm relying on `window._paq.push([function () { ... }])` being processed